### PR TITLE
Allow any identifier to be used as namespace scope

### LINKED ISSUE #9: Doesn't parse custom namespace scopes
Tried to parse:
```thrift
namespace js.ts MyThing
```
And it fails.  It succeeds with `js` instead of `js.ts`

### PR DESCRIPTION
**Pull request by kevinbgreene**
Fixes #9   

* Maybe in codegen we warn if it's not one of the standard scopes?

### CODE REVIEW

**Comment by phated** [comment]
@kevinbgreene the Thrift binary only warns if there isn't a code generator that matches the scope.  For thrift-typescript, we only generate 1 language so I'm not sure it makes sense to warn on other scopes.

File: src/types.ts [hunk 1]
```diff
@@ -79,14 +79,12 @@ export type ConstValue =
...
-  scope: NamespaceScope;
+  scope: Identifier;
   name: Identifier;
 }
 
-export interface NamespaceScope extends SyntaxNode {
-  type: SyntaxType.NamespaceScope;
-  value: '*' | 'cpp' | 'java' | 'py' | 'perl' | 'rb' | 'cocoa' | 'csharp' | 'js'
-}
+export type NamespaceScope =
```
Comment (phated) [inline src/types.ts:86]: I generally think this is unnecessary for our use case but maybe another consumer can benefit from it?

File: src/parser.spec.ts [hunk 2]
```diff
@@ -155,10 +156,59 @@ describe('Parser', () => {
...
+              line: 3,
+              column: 27,
+              index: 51
+            }
+          }
         }
       ]
     };
 
+    console.log('actual: ', JSON.stringify(thrift, null, 2));
```
Comment (phated) [inline src/parser.spec.ts]: Left a console.log in

**Review by phated — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 1 above)
Comment (kevinbgreene) [reply src/types.ts:86]: @phated Yeah, my thought around trying to stick with the scopes as defined by the spec is for the case where a user is starting with thrift for node but work in an environment where they will communicate with other languages. This parser should give them as much guidance as possible and not just worry about the js/ts use case. Leaving this here more for documentation and possible use by another consumer.

**Review by phated — APPROVED** [review]

**Pull request merged**